mailpoet/assets/js/src/settings/store/selectors.ts

```diff
@@ -99,5 +99,8 @@ export function getReEngagementAction(state: State) {
 }
 
 export function hasUnsavedChanges(state: State) {
-  return state.save.hasUnsavedChanges || state.save.inProgress;
+  return (
+    (state.save.hasUnsavedChanges || state.save.inProgress) &&
```

Comment:
Did you consider adding `!_.isEqual(state.data, state.originalData)` into the reducer where we set the `state.save.hasUnsavedChanges` state to true? Theoretically, a selector hook might be called on every rerender of a component where it is used, so having that in the reducer, could be more efficient.

If we decide to keep the comparison of states here in the selector, do we still need to keep tracking the state `state.save.hasUnsavedChanges`?  After you added the comparison between the current and original data, it looks unnecessary and might even be confusing.